src/back-end/lib/resources/proposal/code-with-us.ts

```diff
@@ -149,10 +158,28 @@ const resource: Resource = {
           });
         }
 
+        // Attachments must be validated for both drafts and published opportunities.
+        const validatedAttachments = await validateAttachments(connection, attachments);
+        if (isInvalid(validatedAttachments)) {
+          return invalid({
+            attachments: validatedAttachments.value
+          });
+        }
+
+        // Only validate other fields if not in draft
+        if (validatedStatus.value === CWUProposalStatus.Draft) {
+          return valid({
+            ...request.body,
+            session: request.session,
+            opportunity: validatedCWUOpportunity.value.id,
+            status: validatedStatus.value,
+            attachments: validatedAttachments.value
+          });
+        }
+
         const validatedProposalText = proposalValidation.validateProposalText(proposalText);
         const validatedAdditionalComments = proposalValidation.validateAdditionalComments(additionalComments);
         const validatedProponent = await validateProponent(connection, proponent);
```

Comment:
I think proponents need to be validated somehow for drafts since they are a nested structure?

Reply:
I think it's type-checking for drafts now because `get(body, 'proponent')` in `parseRequestBody` is returning `any`, which will likely lead to a bug.